index.js

```diff
@@ -94,6 +98,9 @@ const rxhr = options => {
         request.onload = onReqLoad
         request.onerror = onReqError
         request.ontimeout = onReqTimeout
+        if (options.progressObserver && options.progressObserver.next) {
```

Comment:
I'd prefer not to force people to use observables to track upload/download progress.

Maybe we can check

```js
typeof options.progressObserver[$$observable] === 'function'
  ? options.progressObserver.next(evt)
  : options.progressObserver(evt)
```

This way we enable poople tu use plain callbacks in case they are not into Observables.

We can also track upload events by doing something like

```js
if (request.upload) {
  request.upload.onprogress = onReqProgress
}

What do you think?
```